src/Altcha.php

```diff
@@ -114,7 +93,13 @@ private static function verifyAndBuildServerSignaturePayload($data): ?ServerSign
             return null;
         }
 
-        return new ServerSignaturePayload($data['algorithm'], $data['verificationData'], $data['signature'], $data['verified']);
+        $algorithm = Algorithm::tryFrom($data['algorithm']);
```

Comment:
Here's where validation of the "user" input happens, so I'm also checking for the value of `algorithm`.